### ISSUE TITLE
Interface elements are missing, pictures are incorrectly provided, tiring etc.

### ISSUE THREAD

**Issue by gosha20777** [AvaloniaUI/Avalonia#5509]
**Describe the bug**
After the update, the interface elements disappear, the images are rendered incorrectly, friezes and tearing.

**To Reproduce**
Update avalonia 0.9.12 -> 0.10.0

https://github.com/lacmus-foundation/lacmus-app/commit/5a6a5bbf6bdc431bf1f68dcefdd6713aeeac35bf

**Expected behavior**
Interface and animations makes slow, tiring, incorrect image rendering

**Screenshots**

*The demonstration: avalonia 0.9.12 (left) vs avalonia 0.10.0 (right)*

![Peek 2021-02-18 13-22](https://user-images.githubusercontent.com/25365364/108343306-d84bd100-71ec-11eb-8266-12296fa76e9e.gif)

![Peek 2021-02-18 13-23](https://user-images.githubusercontent.com/25365364/108343345-e26dcf80-71ec-11eb-898b-b0ff63d9941d.gif)

**Desktop (please complete the following information):**

- OS: Linux (Ubuntu 20.10 / Pop_os 20.10 / Unuttu 20.04)

**Additional context**
```
Thu Feb 18 13:24:02 2021       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 460.39       Driver Version: 460.39       CUDA Version: 11.2     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  Quadro T1000        Off  | 00000000:01:00.0  On |                  N/A |
| N/A   46C    P5     8W /  N/A |    413MiB /  3911MiB |     30%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|    0   N/A  N/A      2016      G   /usr/lib/xorg/Xorg                189MiB |
|    0   N/A  N/A      2827      G   /usr/bin/gnome-shell               76MiB |
|    0   N/A  N/A     13224      G   ...g/netcoreapp5.0/LacmusApp      102MiB | <- avalonia 9.12
|    0   N/A  N/A     13311      G   ...g/netcoreapp5.0/LacmusApp       39MiB | <- avalonia 10.0
+-----------------------------------------------------------------------------+
```

- The new version of Avalonia consumes very little GPU memory. In addition, before and after images are loaded, the amount of memory consumed remains unchanged. For comparison, for avadonia 0.9.12, this amount changes from 40 mb to 150 mb

- The bug with disappearing interface elements also manifests itself in other applications. For example in Citrus.Avalonia.Sandbox

The interface after update is slow and veeeery slow. all animations are slowed down.

In my opinion, something is wrong with GPU usage.

https://github.com/lacmus-foundation/lacmus-app/blob/dev_avalonia_0_10/src/Program.cs#L64 - here are my startup config

I've tryed to use `.UseSkia()` or `.With(new X11PlatformOptions { UseGpu = True })` but nothing helps

**Comment by Gillibald** [comment]
Try to increase https://github.com/AvaloniaUI/Avalonia/blob/master/src/Skia/Avalonia.Skia/SkiaOptions.cs#L23

**Comment by gosha20777** [comment]
@Gillibald can you explain how i can modify this option in `AppBuilder.Configure<App>()`? or something else?

**Comment by gosha20777** [comment]
I tried to add `.With(new SkiaOptions{ MaxGpuResourceSizeBytes = 1024 * 600 * 4 * 12 * 20})` (20x more than in default)

- Interface starts work more faster but
- Pictures are still not rendering correctly
- Interface buttons still disappear

**Comment by grokys** [comment]
Strange, cloning the repo now to take a look.

**Comment by gosha20777** [comment]
@grokys I works with LARGE images
https://disk.yandex.ru/d/KXJ2iF4zUoSSlw - the archive with test image set

**Comment by grokys** [comment]
Thanks, I'm currently running on Windows and with your test image set I indeed see that image rendering is slow, but as you say it's improved by giving skia more GPU resources.

- I'm not seeing the interface buttons disappearing, I guess I need to test on Linux. Have you tried on Windows yourself by any chance?
- What do you mean by "Pictures are still not rendering correctly"? They look OK in the movie you posted above.

In addition, the application here seems to be using the light theme while the movie is using the dark theme, could it be that I'm testing a different commit to you? I checked out `5a6a5bb`.

**Comment by gosha20777** [comment]
@grokys 
1. Commit is correct
2. It doesn't matter what theme you use. You can switch IT in the settings.
  Try to go to File - settings and move your mouse in the settings window:
  
![Peek 2021-02-18 15-30](https://user-images.githubusercontent.com/25365364/108357423-772cf900-71fe-11eb-9bb9-a1fed99533c7.gif)

3. Pictures 
- open pictures (File - Open)
- Select Picture
- Scroll it (mouse wheel)
- See that only part of the picture is shown in front of you, but not all of it:
![Screenshot from 2021-02-18 15-36-47](https://user-images.githubusercontent.com/25365364/108358350-adb74380-71ff-11eb-9cdd-77151ba9b755.png)

**Comment by gosha20777** [comment]
![Peek 2021-02-18 16-06](https://user-images.githubusercontent.com/25365364/108361461-9417fb00-7203-11eb-9bb4-bc02083d2a79.gif)

The sane we can see at citrus.avalonia.sanbox 
https://github.com/worldbeater/Citrus.Avalonia

1. `git clone https://github.com/worldbeater/Citrus.Avalonia.git`
2. build and run Citrus.Avalonia.Sandbox

**Comment by gosha20777** [comment]
- swithed to the default theme from Citrus.Avalonia

Buttons will works fine and now is not disappear

But: images still rendering incorrect: only a small part of image shows.

**Comment by grokys** [comment]
I think only a small part of the image showing is due to how you're using the ZoomAndPan control. If you use DevTools and hover over the `ItemsControl` in the `ZoomBorder` you can see the control highlighted in blue, and the `ItemsControl` doesn't fill the `ZoomBorder`:

![image](https://user-images.githubusercontent.com/1775141/108399049-01368b00-721a-11eb-9965-ee0b57fb3351.png)

(It's a bit hard to see here but there is a blue rectangle overlay displaying the bounds of the control)

It looks like the `Bounds` of the `ItemsControl` is correct but it has a `RenderTransform` applied which is making it smaller. I think the render transform needs to apply to the image not the `ItemsControl` itself.

Given that the problems were:

- Slow: fixed by giving skia more memory
- Buttons disappearing: fixed by switching to default Citrus.Avalonia theme
- Image rendering incorrectly: problem with zooming the wrong thing

I'm going to close this issue. Please add a comment if you think I'm closing in error.

**Comment by maxkatz6** [comment]
> Buttons disappearing: fixed by switching to default Citrus.Avalonia theme

I am not sure if it's normal for Citrus.Avalonia. Problem can be on our side too. @worldbeater

**Comment by gosha20777** [comment]
yep i think too. Citrus.Avalonia is not use any difficult and custom logic, it is simply set of modifiers for default theme...

**Comment by grokys** [comment]
Ok reopening then!

**Comment by worldbeater** [comment]
Yeah encountered yet another runtime theme changing issue with Citrus.Avalonia https://github.com/AvaloniaUI/Avalonia/issues/5530 Assuming [Citrus.Avalonia](https://github.com/worldbeater/Citrus.Avalonia/tree/main/src/Citrus.Avalonia/Theme) is just a bunch of style selectors, probably we are using a weird selector somewhere in the library, although I wasn't able to find out what's causing the buggy behavior.

**Comment by maxkatz6** [comment]
Expected to be fixed in 0.10.11 with updated Skia. It did fix another similar issue on m1.
If not - ping me to reopen.

**Comment by flabbet** [comment]
Seems like there's a regression of this issue in Avalonia 11.3.1

Everything works as expected in `11.3.0`

Observed on Windows with Vulkan backend.

Increasing Skia's GPU `MaxGpuResourceSizeBytes ` doesn't help.

https://github.com/user-attachments/assets/f2c3f53c-1066-41fa-be7e-d9871b3355f6
